node/service_setup.go

```diff
@@ -54,7 +54,7 @@ func (node *Node) setupForBeaconLeader() {
 	nodeConfig, chanPeer := node.initBeaconNodeConfiguration()
 
 	// Register peer discovery service. No need to do staking for beacon chain node.
-	node.serviceManager.RegisterService(service.PeerDiscovery, discovery.New(node.host, nodeConfig, chanPeer))
+	node.serviceManager.RegisterService(service.PeerDiscovery, discovery.New(node.host, nodeConfig, chanPeer, node.AddBeaconPeer))
```

Comment:
BeaconLeader doesn't need add beacon peers

Reply:
this is a generic peer discovery service.  It is needed by everyone. I checked if the node is a beacon node or not within service.